plugins/uefi/fu-uefi-device.c

```diff
@@ -316,18 +316,19 @@ fu_uefi_device_fixup_firmware (FuDevice *device, GBytes *fw, GError **error)
 		fixed_fw = g_bytes_new_from_bytes (fw, 0, fw_length);
 	/* Missing, add a header */
 	} else {
-		guint8 *new_data = g_malloc (fw_length + sizeof(efi_capsule_header_t));
-		guint8 *capsule = new_data + sizeof (efi_capsule_header_t);
+		guint page_size = getpagesize();
+		guint8 *new_data = g_malloc (fw_length + page_size);
+		guint8 *capsule = new_data + page_size;
 		efi_capsule_header_t *header = (efi_capsule_header_t *) new_data;
 
 		g_warning ("missing or invalid embedded capsule header");
 		header->flags = self->capsule_flags;
-		header->header_size = sizeof(efi_capsule_header_t);
-		header->capsule_image_size = fw_length;
+		header->header_size = page_size;
```

Comment:
is `page_size` always >= `sizeof(efi_capsule_header_t)`?

Reply:
Yes, all architectures have a bigger page than the 27 bytes that `efi_capsule_header_t` is.